Makefile

```diff
@@ -120,22 +119,29 @@ fix_cformat:
 $(C_LIB_CARGO_PATH):
 	cargo rustc --release --crate-type cdylib -p qiskit-cext
 
-$(C_QISKIT_H): $(C_LIB_CARGO_PATH)
-	cbindgen --crate qiskit-cext --output $(C_DIR_INCLUDE)/qiskit.h --lang C
+# Compile the C library and move the header into the right location, if it was generated
+$(C_LIB_CARGO_PATH): $(C_DIR_INCLUDE)
+	cargo build --release -p qiskit-cext
+	@if [ -f crates/cext/qiskit.h ]; then mv crates/cext/qiskit.h $(C_DIR_INCLUDE)/qiskit.h; fi
 
 $(C_DIR_LIB):
 	mkdir -p $(C_DIR_LIB)
 
-$(C_LIBQISKIT): $(C_DIR_LIB) $(C_LIB_CARGO_PATH)
+$(C_DIR_INCLUDE):
+	mkdir -p $(C_DIR_INCLUDE)
+
+$(C_LIBQISKIT): $(C_DIR_LIB)  $(C_LIB_CARGO_PATH)
 	cp $(C_LIB_CARGO_PATH) $(C_DIR_LIB)/$(subst _cext,,$(C_LIB_CARGO_FILENAME))
 
-.PHONY: cheader clib c
+$(C_QISKIT_H): $(C_LIB_CARGO_PATH)
+	@if [ ! -f "$(C_QISKIT_H)" ]; then cbindgen --crate qiskit-cext --output $(C_DIR_INCLUDE)/qiskit.h; fi
```

Comment:
There's quite a lot of duplication going on here, and recipes building more than they're specifically asked.  Recipes with specific filenames should build that filename _only_.  Copying the file from one place to another is a separate recipe.

I don't think we should have `cbindgen` as a command-line tool anywhere in the Makefile anymore, since we're now doing this by build script?

Reply:
The header handling is not analogous to how you did the lib handling before: there's separate targets for creating the include dir and copying the header into the include dir. Also `cbindgen` is now no longer needed as command line tool since I added `target/qiskit.h` as tracked file in the build script (to ensure it is rebuilt if it was removed/touched). I tested locally that deleting the headers manually retriggers the build process.